tapis-project/tapis-ui
# task/TUI-307 -- bug fixes tapis training

## Overview:

Fix bugs reported during tapis-training

## Related Github Issues:

- [TUI-307](https://github.com/tapis-project/tapis-ui/issues/307)

## Summary of Changes:

- App Arguments now have undefined values as defaults, as opposed to empty strings. tapis does not accept empty strings for certain app arg fields like description
- Execution requires a jobType, (either batch or fork.) Batch jobs enable selection of a batch queue
- An app that does not already have an execution system listed in the definition can now have one set properly in job submission
- Added step headers to each step
- Renamed Execution System to Execution Options
- Refactor DescriptionList to better handle arrays, sets, empty objects, undefined values and non-string values
- Dedicated JSONDisplay component for tapis-ui/_common that shows JSON objects, with option for simplifying JSON (removing empty arrays and objects, undefined values) and copying to clipboard
- Promote Tabs component to tapis-ui/_common
- SystemDetails now shows both DescriptionList view and JSONDisplay view
- File Browser toolbar now checks permissions on the current path before allowing write operations

## Testing Steps:

1. Test with NoExecSystemOrLogicalQueue v0.0.1 - try changing job type, etc.
2. Add arguments with no description. the job JSON should not show empty string fields for app args
3. Look at system details, make sure all attributes now have a value displayed and that JSON display works correctly in both simplified and unsimplified versions
4. (PENDING) Access a system that you do not have write permissions on and see that the appropriate toolbar buttons are disabled

## UI Photos:

![image](https://user-images.githubusercontent.com/17181582/165590952-ddf1ef0a-6ad9-4258-b0a9-35b08e4b20fe.png)


## Notes:
